src/main/java/talkingbot/task/Task.java

```diff
@@ -0,0 +1,151 @@
+package talkingbot.task;
+
+import java.time.format.DateTimeFormatter;
+
+import talkingbot.exception.TalkingBotException;
+import talkingbot.type.TaskType;
+
+/**
+ * An abstract class for tasks.
+ */
+public abstract class Task {
+
+    /** Formatting of datetime used during entry and saving */
+    protected static final DateTimeFormatter DATE_TIME_ENTRY_FORMAT = DateTimeFormatter
+            .ofPattern("yyyy-MM-dd kkmm");
+
+    /** Formatting of datetime outputs used for displaying to the user */
+    protected static final DateTimeFormatter DATE_TIME_OUT_FORMAT = DateTimeFormatter
+            .ofPattern("MMM dd yyyy kkmm");
+
+    private final String description;
+    private boolean isDone;
+    private final TaskType taskType;
+
+    /**
+     * Constructor for the Task class.
+     *
+     * @param description Description of the task.
+     * @param isDone Marks whether the task has been done.
+     * @param taskType Type of the task.
+     */
+    public Task(String description, boolean isDone, TaskType taskType) {
+        this.description = description;
+        this.isDone = isDone;
+        this.taskType = taskType;
+    }
+
+    /**
+     * Factory method that generates the task object
+     * corresponding to the given type.
+     *
+     * @param fullDescription Full description of the task.
+     * @param type Type of the task.
+     * @return A task: either a Todo, Deadline, or Event.
+     * @throws TalkingBotException If fullDescription is empty, or if any time property is missing.
+     */
+    public static Task generateTask(String fullDescription, String type)
+            throws TalkingBotException {
+        if (type.equals("todo")) {
+            if (fullDescription.isEmpty()) {
+                throw new TalkingBotException("ERROR! todo descriptions cannot be empty"
+                        + " nor only containing whitespaces.");
```

Comment:
Error messages are helpful (points out the error made by the user). Same applies for your other error messages as well